dockerfiles/Dockerfile.source

```diff
@@ -0,0 +1,54 @@
+# --------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------
+# Dockerfile to run ONNXRuntime with source build for CPU
+
+# Ubuntu 16.04 Base Image
+FROM ubuntu:16.04
+WORKDIR /work/deps
+
+# Dependencies: Ubuntu-core
+RUN apt-get update && \
+  apt-get install -y --no-install-recommends sudo \
+    wget \
+    zip \
+    ca-certificates \
+    build-essential curl libcurl4-openssl-dev libssl-dev python3-dev git
+
+# Dependencies: conda
+ENV PATH /opt/miniconda/bin:${PATH}
+RUN wget --quiet https://repo.anaconda.com/miniconda/Miniconda3-4.5.11-Linux-x86_64.sh -O ~/miniconda.sh --no-check-certificate && \
+    /bin/bash ~/miniconda.sh -b -p /opt/miniconda && \
+    rm ~/miniconda.sh && \
+    /opt/miniconda/bin/conda clean -tipsy
+
+RUN conda install -y python=3.6 numpy && \
+    conda clean -aqy && \
+    rm -rf /opt/miniconda/pkgs && \
+    find / -type d -name __pycache__ -prune -exec rm -rf {} \;
+
+# Install the latest cmake
+WORKDIR /code
+RUN wget --quiet https://github.com/Kitware/CMake/releases/download/v3.14.3/cmake-3.14.3-Linux-x86_64.tar.gz
+RUN tar zxf cmake-3.14.3-Linux-x86_64.tar.gz
+
+ENV PATH /code/cmake-3.14.3-Linux-x86_64/bin:${PATH}
+
+# Prepare onnxruntime repository
+RUN git clone --recursive http://github.com/Microsoft/onnxruntime -b master --single-branch
```

Comment:
i think we'll want an ARG which says which TAG or commitID to build from.  It can default to master.
But we want the ability to build from a release tag or specific commit. (Same comment applies to all our Dockerfiles)

Reply:
Applied from PR #1222